data/transactions/logic/README.md

```diff
@@ -40,6 +40,32 @@ has fewer than two elements, the operation fails. Some operations, like
 `frame_dig` and `proto` could fail because of an attempt to access
 above the current stack.
 
+## Stack Types
+
+While every element of the stack is restricted to the types `uint64` and `bytes`, 
+the values of these types may be known to be bounded.  The more common bounded types are 
+named to provide more semantic information in the documentation. They're also used during
+assembly time to do type checking and to provide more informative error messages.
+
+
+#### StackType Definitions
+
+| Name | Bound | AVM Type |
+| ---- | ---- | -------- |
+| uint64 | 0 - 18446744073709551615 | uint64 |
+| none | 0 - 0 | none |
+| name | 1 - 64 | []byte |
+| method | 4 - 4 | []byte |
+| key | 0 - 64 | []byte |
+| bool | 0 - 1 | uint64 |
+| bigint | 0 - 64 | []byte |
+| any | 0 - 0 | any |
+| addr | 32 - 32 | []byte |
+| []byte | 0 - 4096 | []byte |
+| [32]byte | 32 - 32 | []byte |
```

Comment:
maybe just be explicit about the inclusive or exclusive boundaries

Reply:
What do you mean exactly? change the output somehow?

Reply:
also, just pushed some other changes that JJ suggested fyi